source/common/orca/orca_parser.cc

```diff
@@ -0,0 +1,39 @@
+#include "source/common/orca/orca_parser.h"
+
+#include <string>
+
+#include "envoy/http/header_map.h"
+
+#include "source/common/common/base64.h"
+#include "source/common/common/fmt.h"
+
+#include "absl/strings/string_view.h"
+
+using ::Envoy::Http::HeaderMap;
+using ::Envoy::Http::LowerCaseString;
+using xds::data::orca::v3::OrcaLoadReport;
+
+namespace Envoy {
+namespace Orca {
+
+absl::StatusOr<OrcaLoadReport> parseOrcaLoadReportHeaders(const HeaderMap& headers) {
+  OrcaLoadReport load_report;
+
+  // Binary protobuf format.
+  if (const auto header_bin = headers.get(LowerCaseString(kEndpointLoadMetricsHeaderBin));
```

Comment:
We should create a static `LowerCaseString` by the construct on first use to avoid creating the `LowerCaseString` repeatedly.

For example:
```
const Http::LowerCaseString& endpointLoadMetricsHeaderBin() {
  CONSTRUCT_ON_FIRST_USE(Http::LowerCaseString, kEndpointLoadMetricsHeaderBin);
}
```